src/terrain/terrain_base.cpp

```diff
@@ -166,41 +167,36 @@ TerrainBase::stamp_tile_region(
     TerrainOffset x_end = stamp.x_end + x_offset * area_size_ + area_size_ / 2;
     TerrainOffset y_end = stamp.y_end + y_offset * area_size_ + area_size_ / 2;
 
-    for (TerrainOffset x = x_start; x < x_end; x++) {
-        for (TerrainOffset y = y_start; y < y_end; y++) {
-            for (TerrainOffset z = stamp.z_start; z < stamp.z_end; z++) {
-                if (in_range(x, y, z)) {
-                    Tile* tile = get_tile(x, y, z);
-                    if (has_tile_material(stamp.elements_can_stamp, tile)) {
-                        tile->set_material(
-                            biome_.get_material(stamp.mat), stamp.color_id
-                        );
+    if (stamp.elements_can_stamp.has_value()) {
+        for (TerrainOffset x = x_start; x < x_end; x++) {
+            for (TerrainOffset y = y_start; y < y_end; y++) {
+                for (TerrainOffset z = stamp.z_start; z < stamp.z_end; z++) {
+                    if (in_range(x, y, z)) {
+                        Tile* tile = get_tile(x, y, z);
+                        if (has_tile_material(stamp.elements_can_stamp.value(), tile)) {
+                            tile->set_material(
+                                biome_.get_material(stamp.mat), stamp.color_id
+                            );
+                        }
                     }
                 }
             }
         }
-    }
-}
-
-// TODO add this back when want to stamp all all types of materials.
-#if 0
-void
-TerrainBase::stamp_tile_region(
-    int x_start, int y_start, int z_start, int x_end, int y_end, int z_end,
-    const Material* mat, ColorId color_id
-) {
-    // set tiles in region to mat and color_id
-    for (ssize_t x = x_start; x < x_end; x++) {
-        for (ssize_t y = y_start; y < y_end; y++) {
-            for (ssize_t z = z_start; z < z_end; z++) {
-                if (in_range(x, y, z)) {
-                    get_tile(x, y, z)->set_material(mat, color_id);
+    } else {
+        for (TerrainOffset x = x_start; x < x_end; x++) {
+            for (TerrainOffset y = y_start; y < y_end; y++) {
+                for (TerrainOffset z = stamp.z_start; z < stamp.z_end; z++) {
+                    if (in_range(x, y, z)) {
+                        Tile* tile = get_tile(x, y, z);
+                        tile->set_material(
+                            biome_.get_material(stamp.mat), stamp.color_id
+                        );
+                    }
```

Comment:
see above

Reply:
I didn't want to put an if statement inside of a for loop.

Reply:
This is trivially branch predicted. If it's empty, it will stay empty.